cylc/flow/network/scan.py

```diff
@@ -13,302 +13,361 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
-"""Port scan utilities."""
+"""Functionality for searching for workflows running as the current user.
 
+The :py:func:`scan` asynchronous generator yields workflows. Iterate
+over them using an ``async for`` statement::
+
+    async for flow in scan():
+        print(flow['name'])
+
+For further functionality construct a pipe::
+
+    pipe = scan | is_active(True) | contact_info
+    async for flow in pipe:
+        print(f'{flow["name"]} {flow["CYLC_SUITE_HOST"]}')
+
+There are filters which you can you to omit workflows e.g.
+:py:func:`cylc_version` and transformers which acquire more information
+e.g. :py:func:`contact_info`.
+
+.. note: we must manually list functions so they get built into the docs
+
+.. autofunction:: scan
+.. autofunction:: filter_name
+.. autofunction:: is_active
+.. autofunction:: contact_info
+.. autofunction:: cylc_version
+.. autofunction:: api_version
+.. autofunction:: graphql_query
+.. autofunction:: title
+
+"""
+
+from collections.abc import Iterable
 import asyncio
-import os
-from pwd import getpwall
+from pathlib import Path
 import re
-import sys
-import socket
-from cylc.flow import LOG
 
-from cylc.flow.exceptions import SuiteServiceFileError
-import cylc.flow.flags
-from cylc.flow.hostuserutil import is_remote_host, get_host_ip_by_name
+from pkg_resources import (
+    parse_requirements,
+    parse_version
+)
+
+from cylc.flow import LOG
+from cylc.flow.async_util import (
+    pipe,
+    asyncqgen,
+    scandir
+)
 from cylc.flow.network.client import (
     SuiteRuntimeClient, ClientError, ClientTimeout)
+from cylc.flow.exceptions import SuiteStopped
 from cylc.flow.platforms import platform_from_name
 from cylc.flow.suite_files import (
     ContactFileFields,
     SuiteFiles,
-    load_contact_file,
     get_suite_title,
-    get_suite_source_dir
+    load_contact_file_async
 )
 
-DEBUG_DELIM = '\n' + ' ' * 4
-INACTIVITY_TIMEOUT = 10.0
-MSG_QUIT = "QUIT"
-MSG_TIMEOUT = "TIMEOUT"
```

Comment:
This breaks the UIS `master`.

```bash
[I 2020-08-26T11:06:18.893 JupyterHub spawner:1417] Spawning cylc-uiserver --port=57193 -s /home/kinow/Development/python/workspace/cylc-uiserver/../cylc-ui/dist
Failed to set groups [Errno 1] Operation not permitted
Traceback (most recent call last):
  File "/home/kinow/Development/python/workspace/cylc-uiserver/venv/bin/cylc-uiserver", line 11, in <module>
    load_entry_point('cylc-uiserver', 'console_scripts', 'cylc-uiserver')()
  File "/home/kinow/Development/python/workspace/cylc-uiserver/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 489, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/home/kinow/Development/python/workspace/cylc-uiserver/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2852, in load_entry_point
    return ep.load()
  File "/home/kinow/Development/python/workspace/cylc-uiserver/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2443, in load
    return self.resolve()
  File "/home/kinow/Development/python/workspace/cylc-uiserver/venv/lib/python3.7/site-packages/pkg_resources/__init__.py", line 2449, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/kinow/Development/python/workspace/cylc-uiserver/cylc/uiserver/main.py", line 38, in <module>
    from .data_store_mgr import DataStoreMgr
  File "/home/kinow/Development/python/workspace/cylc-uiserver/cylc/uiserver/data_store_mgr.py", line 43, in <module>
    from cylc.flow.network.scan import MSG_TIMEOUT
ImportError: cannot import name 'MSG_TIMEOUT' from 'cylc.flow.network.scan' (/home/kinow/Development/python/workspace/cylc-flow/cylc/flow/network/scan.py)
```

Tested with UIS `master` and sibling PR, it failed to launch the UIS in both cases :cry:

Reply:
https://github.com/cylc/cylc-uiserver/pull/150

Reply:
I need to update that PR, at the moment there will be some issues.